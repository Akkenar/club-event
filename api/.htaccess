<FilesMatch "getRegistrations.php">
  AuthType Basic
  AuthName "Password Protected Area"
  AuthUserFile /home/www/f299233bc103871b19372e6d2c99db54/web/ad-sss/api/.htpasswd
  Require valid-user
</FilesMatch>
