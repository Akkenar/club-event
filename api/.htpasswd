ad-sss:$apr1$G01HZF83$kSFztMQcGPkavFtr07Mng/