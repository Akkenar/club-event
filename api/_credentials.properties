recaptcha.key=TBD
db.host=TBD
db.name=TBD
db.user=TBD
db.password=TBD
email.username=TBD
email.password=TBD
email.host=TBD