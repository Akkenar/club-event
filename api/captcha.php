<?php
require_once './config.php';
require_once './lib/recaptcha-1.2.1/src/autoload.php';

function validateCaptcha($recaptchaResponse)
{
  $config = readConfig();
  if ($config['recaptcha.ignore'] && $recaptchaResponse === 'no-captcha') {
    error_log("Ignoring captcha", 0);
    // By configuration.
    return;
  }

  // Recaptcha utils
  $recaptcha = new \ReCaptcha\ReCaptcha($config['recaptcha.key']);

  // Validate the recaptcha
  $resp = $recaptcha->verify($recaptchaResponse);
  if (!$resp->isSuccess()) {
    $errors = $resp->getErrorCodes();
    http_response_code(500);
    $error_json =
      '{"result": "error", "reason": "' . json_encode($errors) . '"}';
    syslog(LOG_ERR, $error_json);
    echo $error_json;
    die();
  }
}
