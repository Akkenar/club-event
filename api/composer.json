{
  "require": {
    "google/recaptcha": "1.2.1",
    "phpmailer/phpmailer": "5.2.*",
    "pear/html_template_phplib": "1.6.0"
  }
}
