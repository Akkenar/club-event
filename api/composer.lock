{
    "_readme": [
        "This file locks the dependencies of your project to a known state",
        "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies",
        "This file is @generated automatically"
    ],
    "content-hash": "9727a24820ffce14b095b5ecdbddf3b2",
    "packages": [
        {
            "name": "google/recaptcha",
            "version": "1.2.1",
            "source": {
                "type": "git",
                "url": "https://github.com/google/recaptcha.git",
                "reference": "e7add3be59211482ecdb942288f52da64a35f61a"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/google/recaptcha/zipball/e7add3be59211482ecdb942288f52da64a35f61a",
                "reference": "e7add3be59211482ecdb942288f52da64a35f61a",
                "shasum": ""
            },
            "require": {
                "php": ">=5.5"
            },
            "require-dev": {
                "friendsofphp/php-cs-fixer": "^2.2.20|^2.12",
                "php-coveralls/php-coveralls": "^2.1",
                "phpunit/phpunit": "^4.8.36|^5.7.27|^6.59|^7"
            },
            "type": "library",
            "extra": {
                "branch-alias": {
                    "dev-master": "1.2.x-dev"
                }
            },
            "autoload": {
                "psr-4": {
                    "ReCaptcha\\": "src/ReCaptcha"
                }
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "BSD-3-Clause"
            ],
            "description": "Client library for reCAPTCHA, a free service that protects websites from spam and abuse.",
            "homepage": "https://www.google.com/recaptcha/",
            "keywords": [
                "Abuse",
                "captcha",
                "recaptcha",
                "spam"
            ],
            "time": "2018-08-05T09:31:53+00:00"
        },
        {
            "name": "pear/console_getopt",
            "version": "v1.4.1",
            "source": {
                "type": "git",
                "url": "https://github.com/pear/Console_Getopt.git",
                "reference": "82f05cd1aa3edf34e19aa7c8ca312ce13a6a577f"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/pear/Console_Getopt/zipball/82f05cd1aa3edf34e19aa7c8ca312ce13a6a577f",
                "reference": "82f05cd1aa3edf34e19aa7c8ca312ce13a6a577f",
                "shasum": ""
            },
            "type": "library",
            "autoload": {
                "psr-0": {
                    "Console": "./"
                }
            },
            "notification-url": "https://packagist.org/downloads/",
            "include-path": [
                "./"
            ],
            "license": [
                "BSD-2-Clause"
            ],
            "authors": [
                {
                    "name": "Greg Beaver",
                    "email": "cellog@php.net",
                    "role": "Helper"
                },
                {
                    "name": "Andrei Zmievski",
                    "email": "andrei@php.net",
                    "role": "Lead"
                },
                {
                    "name": "Stig Bakken",
                    "email": "stig@php.net",
                    "role": "Developer"
                }
            ],
            "description": "More info available on: http://pear.php.net/package/Console_Getopt",
            "time": "2015-07-20T20:28:12+00:00"
        },
        {
            "name": "pear/html_template_phplib",
            "version": "v1.6.0",
            "source": {
                "type": "git",
                "url": "https://github.com/pear/HTML_Template_PHPLIB.git",
                "reference": "cfea1646644acf65b3e5c2f5f9e30e126e0b21ba"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/pear/HTML_Template_PHPLIB/zipball/cfea1646644acf65b3e5c2f5f9e30e126e0b21ba",
                "reference": "cfea1646644acf65b3e5c2f5f9e30e126e0b21ba",
                "shasum": ""
            },
            "require": {
                "pear/pear-core-minimal": "^1.10.1"
            },
            "require-dev": {
                "phpunit/phpunit": "*"
            },
            "type": "library",
            "autoload": {
                "psr-0": {
                    "HTML": "./"
                }
            },
            "notification-url": "https://packagist.org/downloads/",
            "include-path": [
                "./"
            ],
            "license": [
                "LGPL-2.1"
            ],
            "authors": [
                {
                    "name": "Christian Weiske",
                    "email": "cweiske@php.net",
                    "role": "Lead"
                },
                {
                    "name": "BjÃ¶rn Schotte",
                    "email": "schotte@mayflower.de",
                    "role": "Lead"
                }
            ],
            "description": "preg_* based template system.",
            "homepage": "http://pear.php.net/package/HTML_Template_PHPLIB",
            "time": "2016-01-23T14:35:07+00:00"
        },
        {
            "name": "pear/pear-core-minimal",
            "version": "v1.10.7",
            "source": {
                "type": "git",
                "url": "https://github.com/pear/pear-core-minimal.git",
                "reference": "19a3e0fcd50492c4357372f623f55f1b144346da"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/pear/pear-core-minimal/zipball/19a3e0fcd50492c4357372f623f55f1b144346da",
                "reference": "19a3e0fcd50492c4357372f623f55f1b144346da",
                "shasum": ""
            },
            "require": {
                "pear/console_getopt": "~1.4",
                "pear/pear_exception": "~1.0"
            },
            "replace": {
                "rsky/pear-core-min": "self.version"
            },
            "type": "library",
            "autoload": {
                "psr-0": {
                    "": "src/"
                }
            },
            "notification-url": "https://packagist.org/downloads/",
            "include-path": [
                "src/"
            ],
            "license": [
                "BSD-3-Clause"
            ],
            "authors": [
                {
                    "name": "Christian Weiske",
                    "email": "cweiske@php.net",
                    "role": "Lead"
                }
            ],
            "description": "Minimal set of PEAR core files to be used as composer dependency",
            "time": "2018-12-05T20:03:52+00:00"
        },
        {
            "name": "pear/pear_exception",
            "version": "v1.0.0",
            "source": {
                "type": "git",
                "url": "https://github.com/pear/PEAR_Exception.git",
                "reference": "8c18719fdae000b690e3912be401c76e406dd13b"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/pear/PEAR_Exception/zipball/8c18719fdae000b690e3912be401c76e406dd13b",
                "reference": "8c18719fdae000b690e3912be401c76e406dd13b",
                "shasum": ""
            },
            "require": {
                "php": ">=4.4.0"
            },
            "require-dev": {
                "phpunit/phpunit": "*"
            },
            "type": "class",
            "extra": {
                "branch-alias": {
                    "dev-master": "1.0.x-dev"
                }
            },
            "autoload": {
                "psr-0": {
                    "PEAR": ""
                }
            },
            "notification-url": "https://packagist.org/downloads/",
            "include-path": [
                "."
            ],
            "license": [
                "BSD-2-Clause"
            ],
            "authors": [
                {
                    "name": "Helgi Thormar",
                    "email": "dufuz@php.net"
                },
                {
                    "name": "Greg Beaver",
                    "email": "cellog@php.net"
                }
            ],
            "description": "The PEAR Exception base class.",
            "homepage": "https://github.com/pear/PEAR_Exception",
            "keywords": [
                "exception"
            ],
            "time": "2015-02-10T20:07:52+00:00"
        },
        {
            "name": "phpmailer/phpmailer",
            "version": "v5.2.27",
            "source": {
                "type": "git",
                "url": "https://github.com/PHPMailer/PHPMailer.git",
                "reference": "dde1db116511aa4956389d75546c5be4c2beb2a6"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/PHPMailer/PHPMailer/zipball/dde1db116511aa4956389d75546c5be4c2beb2a6",
                "reference": "dde1db116511aa4956389d75546c5be4c2beb2a6",
                "shasum": ""
            },
            "require": {
                "ext-ctype": "*",
                "php": ">=5.0.0"
            },
            "require-dev": {
                "doctrine/annotations": "1.2.*",
                "jms/serializer": "0.16.*",
                "phpdocumentor/phpdocumentor": "2.*",
                "phpunit/phpunit": "4.8.*",
                "symfony/debug": "2.8.*",
                "symfony/filesystem": "2.8.*",
                "symfony/translation": "2.8.*",
                "symfony/yaml": "2.8.*",
                "zendframework/zend-cache": "2.5.1",
                "zendframework/zend-config": "2.5.1",
                "zendframework/zend-eventmanager": "2.5.1",
                "zendframework/zend-filter": "2.5.1",
                "zendframework/zend-i18n": "2.5.1",
                "zendframework/zend-json": "2.5.1",
                "zendframework/zend-math": "2.5.1",
                "zendframework/zend-serializer": "2.5.*",
                "zendframework/zend-servicemanager": "2.5.*",
                "zendframework/zend-stdlib": "2.5.1"
            },
            "suggest": {
                "league/oauth2-google": "Needed for Google XOAUTH2 authentication"
            },
            "type": "library",
            "autoload": {
                "classmap": [
                    "class.phpmailer.php",
                    "class.phpmaileroauth.php",
                    "class.phpmaileroauthgoogle.php",
                    "class.smtp.php",
                    "class.pop3.php",
                    "extras/EasyPeasyICS.php",
                    "extras/ntlm_sasl_client.php"
                ]
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "LGPL-2.1"
            ],
            "authors": [
                {
                    "name": "Jim Jagielski",
                    "email": "jimjag@gmail.com"
                },
                {
                    "name": "Marcus Bointon",
                    "email": "phpmailer@synchromedia.co.uk"
                },
                {
                    "name": "Andy Prevost",
                    "email": "codeworxtech@users.sourceforge.net"
                },
                {
                    "name": "Brent R. Matzelle"
                }
            ],
            "description": "PHPMailer is a full-featured email creation and transfer class for PHP",
            "time": "2018-11-15T22:32:31+00:00"
        }
    ],
    "packages-dev": [],
    "aliases": [],
    "minimum-stability": "stable",
    "stability-flags": [],
    "prefer-stable": false,
    "prefer-lowest": false,
    "platform": [],
    "platform-dev": []
}
