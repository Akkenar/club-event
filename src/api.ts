import environment from './environment/environment';

export function getApi(): any {
  // Depending on whether the file is replaced by Webpack, the api URLs
  // will change based on the type of build. See `environment.ts` and
  // `environment.prod.ts` for more details.
  return environment.api;
}
