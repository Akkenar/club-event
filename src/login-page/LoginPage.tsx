import * as React from 'react';
import { useContext, useEffect } from 'react';
import getKey from '../core/intl/getKey';
import LanguageContext from '../core/intl/LanguageContext';
import { goToTop, setPageTitle } from '../core/page.lib';
import LoginForm from './LoginForm';

const LoginPage = () => {
  const { messages } = useContext(LanguageContext);

  useEffect(() => {
    setPageTitle(getKey('login.page.title', messages));
    goToTop();
  });

  return (
    <div className="LoginPage">
      <LoginForm />
    </div>
  );
};

export default LoginPage;
