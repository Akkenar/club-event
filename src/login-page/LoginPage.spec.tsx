import 'jest-dom/extend-expect';
import * as React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, wait, waitForElement } from 'react-testing-library';
import { Results } from '../core/api.type';
import {
  mockGrecaptcha,
  mockResponse,
  setValidInputData,
} from '../test-utils/test-utils.lib';
import LoginPage from './LoginPage';

mockGrecaptcha();

const LoginPageWithRouter = () => {
  return (
    <BrowserRouter>
      <LoginPage />
    </BrowserRouter>
  );
};

describe('LoginPage', () => {
  it('should match snapshot', () => {
    const wrapper = render(<LoginPageWithRouter />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should submit the login', async () => {
    const wrapper = render(<LoginPageWithRouter />);
    mockResponse(Results.SUCCESS);
    setValidInputData({ username: 'Foo', password: 'Bar' }, wrapper);

    // Submit the form!
    wrapper.queryByText('login.form.submit').click();

    // First wait for the loader to be displayed
    await waitForElement(() => wrapper.queryByTestId('loader'));

    // Wait for the loader to be dismissed
    await wait(() =>
      expect(wrapper.queryByTestId('loader')).not.toBeInTheDocument()
    );

    // Should have been redirected.
  });

  it('should handle login error', async () => {
    const wrapper = render(<LoginPageWithRouter />);
    mockResponse(Results.ERROR, 'Error');
    setValidInputData({ username: 'Foo', password: 'Bar' }, wrapper);

    // Submit the form!
    wrapper.queryByText('login.form.submit').click();

    // First wait for the loader to be displayed
    await waitForElement(() => wrapper.queryByTestId('loader'));

    // Wait for the loader to be dismissed
    await wait(() =>
      expect(wrapper.queryByTestId('loader')).not.toBeInTheDocument()
    );

    expect(wrapper.queryByTestId('backend-error')).toHaveClass('visible');
  });
});
