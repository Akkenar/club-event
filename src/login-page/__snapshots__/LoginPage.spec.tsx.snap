// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match snapshot 1`] = `
<body>
  <div>
    <div
      class="LoginPage"
    >
      <div
        class="ui segment"
      >
        <h1
          class="ui header"
        >
          login.page.title
        </h1>
        <form
          class="ui form LoginForm"
          novalidate=""
        >
          <div
            class="field"
          >
            <label
              for="username"
            >
              login.form.username
            </label>
            <input
              aria-invalid="false"
              autocomplete="username"
              data-testid="username"
              id="username"
              maxlength="255"
              name="username"
              required=""
              value=""
            />
          </div>
          <div
            class="field"
          >
            <label
              for="password"
            >
              login.form.password
            </label>
            <input
              aria-invalid="false"
              autocomplete="current-password"
              data-testid="password"
              id="password"
              maxlength="255"
              name="password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="ui error message"
            data-testid="backend-error"
          />
          <div
            style="height: 78px; width: 302px;"
          >
            loading
          </div>
          <div
            class="LoginForm__submit-button-container"
          >
            <button
              class="ui black button full-width LoginForm__submit-button"
              type="submit"
            >
              login.form.submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;
